models/demos/t3000/falcon40b/tt/falcon_mlp.py

```diff
@@ -108,84 +77,73 @@ def __call__(
             assert False
 
     def fwd_decode(self, x: List[ttnn.experimental.tensor.Tensor]) -> List[ttnn.experimental.tensor.Tensor]:
-        hidden_states = []
-        for i in range(len(x)):
-            hidden_states.append(
-                ttnn.experimental.operations.primary.matmul_1d(
-                    x[i],
-                    self.dense_h_to_4h_weights[i],
-                    program_config=self.model_config["DENSE_H_TO_4H_MM_PROGCFG"],
-                    output_mem_config=self.model_config["DENSE_H_TO_4H_MM_OUTPUT_MEMCFG"],
-                    output_dtype=self.model_config["DENSE_H_TO_4H_MM_OUTPUT_DTYPE"],
-                    compute_kernel_config=self.model_config["COMPUTE_KERNEL_CONFIG"],
-                )
-            )
-            x[i].deallocate(True)
-        for i in range(len(hidden_states)):
-            hidden_states[i] = ttnn.experimental.tensor.sharded_to_interleaved(
-                hidden_states[i], output_mem_config=self.model_config["DEFAULT_MEMCFG"]
-            )
+
+        hidden_states = ttnn.experimental.operations.primary.matmul_1d(
+            x,
+            self.dense_h_to_4h_weights,
+            program_config=self.model_config["DENSE_H_TO_4H_MM_PROGCFG"],
+            output_mem_config=self.model_config["DENSE_H_TO_4H_MM_OUTPUT_MEMCFG"],
+            output_dtype=self.model_config["DENSE_H_TO_4H_MM_OUTPUT_DTYPE"],
+            compute_kernel_config=self.model_config["COMPUTE_KERNEL_CONFIG"],
+        )
+        x.deallocate(True)
+
+        hidden_states = ttnn.experimental.tensor.sharded_to_interleaved(
+            hidden_states, output_mem_config=self.model_config["DEFAULT_MEMCFG"]
+        )
         hidden_states = ttnn.all_gather(
```

Comment:
@ayerofieiev-tt, here is one of examples of all_gather for multi-device tensors.